fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StringLikeLiteral.java

```
@@ -62,6 +66,59 @@ public String getValue() {
         return value;
     }
 
+    @Override
+    public LiteralExpr toLegacyLiteral() {
+        return new org.apache.doris.analysis.StringLiteral(value);
+    }
+
+    @Override
+    public int compareTo(ComparableLiteral other) {
+        if (other instanceof StringLikeLiteral) {
+            // compare string with utf-8 byte array, same with DM,BE,StorageEngine
+            byte[] thisBytes = null;
+            byte[] otherBytes = null;
+            try {
+                thisBytes = getStringValue().getBytes("UTF-8");
+                otherBytes = ((Literal) other).getStringValue().getBytes("UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                Preconditions.checkState(false);
```

do not eat the exception. new AnalysisException and wrap UnsupportedEncodingException and then throw it

---

> do not eat the exception. new AnalysisException and wrap UnsupportedEncodingException and then throw it

fix